```
[2020-01-22T16:33:28.347Z] /home/ec2-user/workspace/test_rosdistro_fuse_PR-134/package/fuse_constraints/test/test_variable_constraints.cpp:117:3:   required from here

[2020-01-22T16:33:28.347Z] /usr/src/googletest/googletest/include/gtest/gtest.h:1392:11: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]

[2020-01-22T16:33:28.347Z]    if (lhs == rhs) {
```
```
EXPECT_EQ(expected0.size(), std::distance(actual0.begin(), actual0_iter));
```
`std::distance` returns a signed value...which is slightly unexpected to me.

The xenial tests passed, but not the bionic tests. I'm assuming the later compiler version is more strict.

--- a/fuse_constraints/test/test_variable_constraints.cpp
+++ b/fuse_constraints/test/test_variable_constraints.cpp
@@ -114,10 +114,10 @@ TEST(VariableConstraints, GetConstraints)
   auto actual3_iter = actual3.begin();
   actual3_iter = vars.getConstraints(3u, actual3_iter);
 
-  EXPECT_EQ(expected0.size(), std::distance(actual0.begin(), actual0_iter));
-  EXPECT_EQ(expected1.size(), std::distance(actual1.begin(), actual1_iter));
-  EXPECT_EQ(expected2.size(), std::distance(actual2.begin(), actual2_iter));
-  EXPECT_EQ(expected3.size(), std::distance(actual3.begin(), actual3_iter));
+  EXPECT_EQ(static_cast<std::ptrdiff_t>(expected0.size()), std::distance(actual0.begin(), actual0_iter));
+  EXPECT_EQ(static_cast<std::ptrdiff_t>(expected1.size()), std::distance(actual1.begin(), actual1_iter));
+  EXPECT_EQ(static_cast<std::ptrdiff_t>(expected2.size()), std::distance(actual2.begin(), actual2_iter));
+  EXPECT_EQ(static_cast<std::ptrdiff_t>(expected3.size()), std::distance(actual3.begin(), actual3_iter));
 }
 
 int main(int argc, char **argv)

